dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -338,91 +348,95 @@ public Group createWorkflowGroup(Context context, Collection collection, int ste
         // Check authorisation - Must be an Admin to create Workflow Group
         AuthorizeUtil.authorizeManageWorkflowsGroup(context, collection);
 
-        if (getWorkflowGroup(collection, step) == null) {
+        if (getWorkflowGroup(context, collection, step) == null) {
             //turn off authorization so that Collection Admins can create Collection Workflow Groups
             context.turnOffAuthorisationSystem();
             Group g = groupService.create(context);
-            context.restoreAuthSystemState();
-
             groupService.setName(g,
                                  "COLLECTION_" + collection.getID() + "_WORKFLOW_STEP_" + step);
             groupService.update(context, g);
+            context.restoreAuthSystemState();
             setWorkflowGroup(context, collection, step, g);
-
         }
 
-        return getWorkflowGroup(collection, step);
+        return getWorkflowGroup(context, collection, step);
     }
 
     @Override
     public void setWorkflowGroup(Context context, Collection collection, int step, Group group)
         throws SQLException, AuthorizeException {
-        // we need to store the old group to be able to revoke permissions if granted before
-        Group oldGroup = null;
-        int action;
+        Workflow workflow = null;
+        try {
+            workflow = workflowFactory.getWorkflow(collection);
+        } catch (IOException | WorkflowConfigurationException e) {
+            log.error(LogManager.getHeader(context, "setWorkflowGroup",
+                    "collection_id=" + collection.getID() + " " + e.getMessage()), e);
+        }
+        if (!StringUtils.equals("default", workflow.getID())) {
```

Comment:
Should this hardcoded string be a constant?

Reply:
done

Reply:
I'm still seeing the string "default" here.  Should that become a constant?  I guess I'm not sure how/when a Workflow ID gets set to the string "default"

Reply:
It is the name of the workflow configured in the workflow.xml to mimic the legacy workflow. I have introduced a constant